logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -41,6 +41,10 @@ def register
     @strategy.register
   end
 
+  def do_register(dlq_manager=nil)
```

Comment:
Can we add a comment here why we are doing this in a separate method vs in the constructor? Will be helpful context.